cmd/kubeadm/app/constants/constants_test.go

```diff
@@ -182,28 +186,30 @@ func TestEtcdSupportedVersion(t *testing.T) {
 		},
 	}
 	for _, rt := range tests {
-		actualVersion, actualError := EtcdSupportedVersion(rt.kubernetesVersion)
-		if actualError != nil {
-			if rt.expectedError == nil {
-				t.Errorf("failed EtcdSupportedVersion:\n\texpected no error, but got: %v", actualError)
-			} else if actualError.Error() != rt.expectedError.Error() {
-				t.Errorf(
-					"failed EtcdSupportedVersion:\n\texpected error: %v\n\t  actual error: %v",
-					rt.expectedError,
-					actualError,
-				)
-			}
-		} else {
-			if rt.expectedError != nil {
-				t.Errorf("failed EtcdSupportedVersion:\n\texpected error: %v, but got no error", rt.expectedError)
-			} else if strings.Compare(actualVersion.String(), rt.expectedVersion.String()) != 0 {
-				t.Errorf(
-					"failed EtcdSupportedVersion:\n\texpected version: %s\n\t  actual version: %s",
-					rt.expectedVersion.String(),
-					actualVersion.String(),
-				)
+		t.Run(rt.kubernetesVersion, func(t *testing.T) {
```

Comment:
I might prefer giving these test cases names describing what they are testing, but i'm not terribly concerned about it

Reply:
I think the test output looks quite understandable:

```
$ go test -run TestEtcdSupportedVersion
=== RUN   TestEtcdSupportedVersion
=== RUN   TestEtcdSupportedVersion/1.99.0
=== RUN   TestEtcdSupportedVersion/1.10.2
=== RUN   TestEtcdSupportedVersion/1.11.0
=== RUN   TestEtcdSupportedVersion/1.12.1
=== RUN   TestEtcdSupportedVersion/1.13.1
=== RUN   TestEtcdSupportedVersion/1.14.0
--- PASS: TestEtcdSupportedVersion (0.00s)
    --- PASS: TestEtcdSupportedVersion/1.99.0 (0.00s)
    --- PASS: TestEtcdSupportedVersion/1.10.2 (0.00s)
    --- PASS: TestEtcdSupportedVersion/1.11.0 (0.00s)
    --- PASS: TestEtcdSupportedVersion/1.12.1 (0.00s)
    --- PASS: TestEtcdSupportedVersion/1.13.1 (0.00s)
    --- PASS: TestEtcdSupportedVersion/1.14.0 (0.00s)
```

Adding something like "testing etc supported version 1.10.2" wouldn't add much content here in my opinion.

Reply:
Yep, the output looks good. I like to think the name is more about what is this test case actually testing, or the assertion being made. The name "1.14.0" doesn't mean much to me, but I could certainly trace it back and figure it out which is more than enough imo.

Reply:
great! lgtm then :) ?